src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.spec.ts

```diff
@@ -159,89 +192,137 @@ describe('DotEditPageToolbarComponent', () => {
         fixture.detectChanges();
 
         let pageStateResult;
-        component.pageState.subscribe((res) => {
+        component.changeState.subscribe((res) => {
             pageStateResult = res;
         });
 
-        let lockPageResult;
-        component.lockPage.subscribe((res) => {
-            lockPageResult = res;
-        });
-
         clickStateButton('edit');
 
-        expect(component.pageLocked).toBe(true, 'lock page');
-        expect(lockPageResult).toBe(true, 'lock page output emitted');
-        expect(pageStateResult).toEqual(PageState.EDIT, 'page state output emitted');
+        expect(component.page.locked).toBe(true, 'lock page');
+        expect(pageStateResult).toEqual({mode: PageMode.EDIT, lock: true}, 'page state output emitted');
     });
 
     it('should go to preview if user unlock the page while is in edit', () => {
         fixture.detectChanges();
 
         // Set the page locked and in edit mode
         clickLocker();
-        expect(component.pageLocked).toBe(true, 'locked page');
-        expect(component.stateSelected).toEqual(PageState.EDIT, 'edit mode');
+        expect(component.page.locked).toBe(true, 'locked page');
+        expect(component.stateSelected).toEqual(PageMode.EDIT, 'edit mode');
 
         clickLocker();
-        expect(component.pageLocked).toBe(false, 'unlocked page');
-        expect(component.stateSelected).toEqual(PageState.PREVIEW, 'preview mode');
+        expect(component.page.locked).toBe(false, 'unlocked page');
+        expect(component.stateSelected).toEqual(PageMode.PREVIEW, 'preview mode');
     });
 
     it('should go to edit if user lock the while is in preview', () => {
         fixture.detectChanges();
 
         clickLocker();
-        expect(component.pageLocked).toBe(true, 'locked page');
-        expect(component.stateSelected).toEqual(PageState.EDIT, 'edit mode');
+        expect(component.page.locked).toBe(true, 'locked page');
+        expect(component.stateSelected).toEqual(PageMode.EDIT, 'edit mode');
     });
 
-    describe('moving through page states', () => {
+    it('should copy to clipboard url', () => {
+        spyOn(component, 'copyUrlToClipboard');
+        fixture.detectChanges();
+
+        const copyUrlButton: DebugElement = de.query(By.css('.edit-page-toolbar__copy-url'));
+
+        copyUrlButton.nativeElement.click();
+
+        expect(component.copyUrlToClipboard).toHaveBeenCalledTimes(1);
```

Comment:
this method just check the the 'copyUrlToClipboard' method is call, but dont check f this is implement right
I think this test is not neccesary